src/databricks/labs/ucx/assessment/aws.py

```diff
@@ -206,6 +213,137 @@ def get_role_policy(self, role_name, policy_name: str | None = None, attached_po
                     policy_actions.append(AWSPolicyAction("s3", privilege, f"s3a://{match.group(1)}"))
         return policy_actions
 
+    def add_uc_role(self, role_name, policy_name, s3_prefixes: set[str], account_id: str, kms_key=None):
+        aws_role_trust_doc: dict = {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Principal": {
+                        "AWS": "arn:aws:iam::414351767826:role/unity-catalog-prod-UCMasterRole-14S5ZJVKOTYTL"
+                    },
+                    "Action": "sts:AssumeRole",
+                    "Condition": {"StringEquals": {"sts:ExternalId": "0000"}},
```

Comment:
External id must be a variable. Is it account id or credentials id?

Reply:
That's how we set it. We need to set it and then reset it once we capture it from the credential.
It is a strange process documented here:
https://docs.databricks.com/en/connect/unity-catalog/storage-credentials.html

Reply:
Okay, add this as an explicit code comment near this line, so that we avoid the same reaction from the next person reading this

Reply:
Done